src/components/header/Header.tsx

```diff
@@ -0,0 +1,89 @@
+/// <reference types="vite-plugin-svgr/client" />
+import React, { Suspense, useCallback } from "react";
+import AppIcon from "@/assets/app-icon.svg?react";
+import ArrowDown from "@/assets/arrow-down.svg?react";
+import Typography from "@/components/typography/Typography";
+import Breadcrumb, { BreadcrumbItem } from "./breadcrumb/Breadcrumb";
+import { APP_NAME } from "@/common/consts";
+import { useRouter } from "@tanstack/react-router";
+import { useAuth } from "@/hooks/useAuth";
+import { lazy } from "react";
+
+type HeaderProps = {
+  breadcrumbs: BreadcrumbItem[];
+  user: string;
+};
+
+function DynamicLoader(thumbnail: number) {
+  const MarkdownPreview = lazy(
+    () =>
+      import(`../../assets/account-thumbnails/color-${thumbnail}.svg?react`),
+  );
+  return (
+    <Suspense fallback={<></>}>
+      <MarkdownPreview />
+    </Suspense>
+  );
+}
+
+const Header = React.memo(({ breadcrumbs, user }: HeaderProps) => {
+  const router = useRouter();
+  const { signOut, thumbnail } = useAuth();
+
+  const getFirstLetter: () => string = useCallback(() => {
+    const match = user.match(/([a-zA-Z])/);
+    return match ? match[1] : "";
+  }, [user]);
+
+  return (
+    <div className="w-full text-left bg-mono/basic-16 h-17 items-center justify-between gap-4 flex">
+      <nav aria-label="breadcrumb">
+        <div className="flex w-fit flex-wrap items-center rounded-md bg-blue-gray-50 bg-opacity-60 py-2 pl-5.5">
+          <Typography>
+            <AppIcon style={{ width: "32.26px", height: "32.26px" }} />
+          </Typography>
+          <Typography
+            variant="headline-sm"
+            className="text-mono/basic-4 pl-2.5 pr-4.25"
+          >
+            {APP_NAME}
+          </Typography>
+          <div className="flex cursor-pointer items-center antialiased transition-colors duration-100 bg-mono/basic-14 rounded-full h-fit">
+            <div className="flex items-center py-2.25 pl-4 pr-5">
+              {breadcrumbs.map((breadcrumb) => (
+                <Breadcrumb breadcrumb={breadcrumb} key={breadcrumb.text} />
+              ))}
+            </div>
+          </div>
+        </div>
+      </nav>
+      <div className="place-self-end flex items-center h-full pr-5.5">
+        <div className="relative flex justify-center items-center">
+          {DynamicLoader(thumbnail)}
+          <div className="text-white absolute inset-0 flex justify-center items-center">
+            {getFirstLetter().toUpperCase()}
+          </div>
+        </div>
+        <div
+          className="group flex items-center cursor-pointer"
+          onClick={async () => {
```

Comment:
It's better to encapsulate the `router.invalidate()` inside the `signOut` function, as everytime you would signOut you would like it to happen (I guess)

Reply:
Also why is it async?